src/Nowruz/modules/credentials/KYB/index.tsx

```diff
@@ -15,6 +15,11 @@ export const KYBModal: React.FC<KYBModalProps> = ({ open, setOpen }) => {
     uplaod: false,
     success: false,
   });
+
+  useEffect(() => {
```

Comment:
every time you open or close the modal this `useEffect` runs, why you need it? 
this is initial value and when you open Modal every time, you have initial value.
i mean why forced to be initial value?

Reply:
closing and opening KYBModal doesn't initialize the state value, which was weird for me too. I added useEffect to meke component to listen to property change. If you remove useEffect, opening desc modal doesn't work properly.

Reply:
i checked your code and it totally works fine.
you open KYB modal via `openVerifiyAlert` from outside and you wanna every time you set `openVerifiyAlert` to true the description modal is opened.
so your initial state is true
but after closing in `handleClose` you should restart to initial value (because closing KYB modal is handled via `openVerifiyAlert` not `openModals` inside)
`const handleClose = () => {
    setopenModals({ desc: true, upload: false, success: false }); // initial state
    setOpen(false);
  };`
  
  whenever you have this situation you should handle all states from outside, debugging is more easier.